pkgs/tools/misc/rpcsvc-proto/default.nix

```diff
@@ -11,12 +11,27 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-DXRLEpOanWcxCSTC8OxEw5SWKpgFz7oTPukzfnPrAOk=";
   };
 
+  patches = [
+    # https://github.com/thkukuk/rpcsvc-proto/pull/14
+    (fetchpatch {
+      name = "follow-RPCGEN_CPP-env-var";
+      url = "https://github.com/thkukuk/rpcsvc-proto/commit/e772270774ff45172709e39f744cab875a816667.diff";
+      sha256 = "sha256-KrUD6YwdyxW9S99h4TB21ahnAOgQmQr2tYz++MIbk1Y=";
+    })
+  ];
+
   outputs = [ "out" "man" ];
 
-  depsBuildBuild = [ buildPackages.stdenv.cc ];
   nativeBuildInputs = [ autoreconfHook ];
 
-  postPatch = lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
+  RPCGEN_CPP = "${stdenv.cc.targetPrefix}cpp";
+
+  postPatch = ''
+    # replace fallback cpp with the target prefixed cpp
+    substituteInPlace rpcgen/rpc_main.c \
+      --replace 'CPP = "cpp"' \
+                'CPP = "${targetPackages.stdenv.cc.targetPrefix}cpp"'
+  '' + lib.optionalString (stdenv.hostPlatform != stdenv.buildPlatform) ''
```

Comment:
How about we do this always? I hope this would prevent cross regressions when updating.

Reply:
When host == build there won't be a buildPackages.rpcsvc-proto

However the conditional can be changed to canExecute